runtime/near-runtime-fees/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+//! Describes the various costs incurred by creating receipts.
+//! We use the following abbreviation for readability:
+//! * sir -- sender is receiver. Receipts that are directed by an account to itself are guaranteed
+//!   to no be cross-shard which is cheaper than cross-shard;
+use serde::{Deserialize, Serialize};
+pub type Gas = u64;
+
+/// Costs associated with an object that can only be sent over the network (and executed
+/// by the receiver).
+#[derive(Debug, Serialize, Deserialize, Clone, Hash, PartialEq, Eq)]
+pub struct Fee {
+    /// Fee for sending an object from the sender to itself, guaranteeing that it does not leave
+    /// the shard.
+    send_sir: Gas,
+    /// Fee for sending an object potentially across the shards.
+    send_not_sir: Gas,
+    /// Fee for executing the object.
+    execution: Gas,
+}
+
+impl Fee {
+    pub fn send_fee(&self, sir: bool) -> Gas {
+        if sir {
+            self.send_sir
+        } else {
+            self.send_not_sir
+        }
+    }
+
+    pub fn exec_fee(&self) -> Gas {
+        self.execution
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Hash, PartialEq, Eq)]
+pub struct RuntimeFeesConfig {
+    /// Describes the cost of creating an action receipt, `ActionReceipt`, excluding the actual cost
+    /// of actions.
+    pub action_receipt_creation_config: Fee,
+    /// Describes the cost of creating a data receipt, `DataReceipt`.
+    pub data_receipt_creation_config: DataReceiptCreationConfig,
+    /// Describes the cost of creating a certain action, `Action`. Includes all variants.
+    pub action_creation_config: ActionCreationConfig,
+}
+
+/// Describes the cost of creating a data receipt, `DataReceipt`.
+#[derive(Debug, Serialize, Deserialize, Clone, Hash, PartialEq, Eq)]
+pub struct DataReceiptCreationConfig {
+    /// Base cost of creating a data receipt.
+    pub base_cost: Fee,
+    /// Additional cost per byte sent.
+    pub cost_per_byte: Fee,
+}
+
+/// Describes the cost of creating a specific action, `Action`. Includes all variants.
+#[derive(Debug, Serialize, Deserialize, Clone, Hash, PartialEq, Eq)]
+pub struct ActionCreationConfig {
+    /// Base cost of creating an account.
+    pub create_account_cost: Fee,
+
+    /// Base cost of deploying a contract.
+    pub deploy_contract_cost: Fee,
+    /// Cost per byte of deploying a contract.
+    pub deploy_contract_cost_per_byte: Fee,
+
+    /// Base cost of calling a function.
+    pub function_call_cost: Fee,
+    /// Cost per byte of method name and arguments of calling a function.
+    pub function_call_cost_per_byte: Fee,
+
+    /// Base cost of making a transfer.
+    pub transfer_cost: Fee,
+
+    /// Base cost of staking.
+    pub stake_cost: Fee,
+
+    /// Base cost of adding a key.
+    pub add_key_cost: Fee,
```

Comment:
Also need `access_key_config` to calculate cost of creating a given access key.
It's should be a config for each permission:
```rust
pub struct AccessKeyCreationConfig {
    /// Base cost of full access permission.
    pub full_access_cost: Fee,

    /// Base cost of function call permission.
    pub function_call_cost: Fee,
    /// Cost per byte of method names and receiver_id.
    pub function_call_cost_per_byte: Fee,
}
```
```

Reply:
I suggest, for consistency, we limit the size of the method name that can be called from access key similarly to limiting account id name and then we charge a fixed rate for this operation.

Reply:
Btw, this is already included in here: https://github.com/nearprotocol/nearcore/blob/cc195294c733882a947459b59c6313e60c6e7718/runtime/near-runtime-fees/src/lib.rs#L79

Reply:
Btw this is already done here: https://github.com/nearprotocol/nearcore/blob/cc195294c733882a947459b59c6313e60c6e7718/runtime/near-runtime-fees/src/lib.rs#L79